libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -1104,6 +1134,8 @@ __completion_mechanism __dispatch_memcpy_async_global_to_shared(_Group const & _
             }
         }
         // Fallthrough..
+    ),(
+        __cuda_ptx_cp_async_shared_global_is_not_supported_before_SM_80__();
```

Comment:
```suggestion
    NV_IF_TARGET(NV_PROVIDES_SM_80, (
        if _LIBCUDACXX_CONSTEXPR_AFTER_CXX14 (_Align >= 4) {
            const bool __can_use_async_group = __allowed_completions & uint32_t(__completion_mechanism::__async_group);
            if (__can_use_async_group) {
                __cp_async_shared_global_mechanism<_Align>(__group, __dest_char, __src_char, __size);
                return __completion_mechanism::__async_group;
            }
        }
        // Fallthrough..
    ));
```

For some reason, multi-line suggetsions are broken for me. Here, you should really fall through to the fallback mechanism though (and not trigger the linker error).